examples/remote-settings-cli/src/downloader.rs

```diff
@@ -0,0 +1,289 @@
+use anyhow::{Context, Result};
+use futures::{stream::FuturesUnordered, StreamExt};
+use indicatif::{MultiProgress, ProgressBar, ProgressStyle};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::{collections::HashMap, path::PathBuf, sync::Arc};
+
+use crate::git_ops::GitOps;
+
+const REMOTE_SETTINGS_URL: &str = "https://firefox.settings.services.mozilla.com/v1";
+const DEFAULT_OUTPUT_DIR: &str = "components/remote_settings";
+const COLLECTIONS: &[&str] = &[
+    "anti-tracking-url-decoration",
+    "cookie-banner-rules-list",
+    "devtools-compatibility-browsers",
+    "devtools-devices",
+    "hijack-blocklists",
+    "language-dictionaries",
+    "password-recipes",
+    "password-rules",
+    "search-config-icons",
+    "search-config-v2",
+    "search-config-overrides-v2",
+    "search-telemetry-v2",
+    "sites-classification",
+    "top-sites",
+    "translations-models",
+    "translations-wasm",
+    "url-classifier-skip-urls",
+    "url-parser-default-unknown-schemes-interventions",
+    "urlbar-persisted-search-terms",
+    "websites-with-shared-credential-backends",
+];
+
+#[derive(Debug, Serialize, Deserialize)]
+pub(crate) struct ServerResponse {
+    pub(crate) data: Vec<Value>,
+    pub(crate) timestamp: u64,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub(crate) struct PackagedData {
+    pub(crate) collections: HashMap<String, CollectionData>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub(crate) struct CollectionData {
+    pub(crate) timestamp: u64,
+    pub(crate) records: Vec<Value>,
+}
+
+pub(crate) struct CollectionDownloader {
+    pub(crate) client: reqwest::Client,
+    pub(crate) multi_progress: Arc<MultiProgress>,
+    pub(crate) output_dir: PathBuf,
+}
+
+impl CollectionDownloader {
+    pub(crate) fn new(root_path: PathBuf) -> Self {
+        let output_dir = root_path.join(DEFAULT_OUTPUT_DIR);
+        Self {
+            client: reqwest::Client::new(),
+            multi_progress: Arc::new(MultiProgress::new()),
+            output_dir,
+        }
+    }
+
+    async fn fetch_collection(
+        &self,
+        collection: String,
+        pb: ProgressBar,
+    ) -> Result<(String, CollectionData)> {
+        // First get the last_modified from monitor
+        let monitor_url = format!(
+            "{}/buckets/monitor/collections/changes/records",
+            REMOTE_SETTINGS_URL,
+        );
+
+        let monitor_response: Value = self.client.get(&monitor_url).send().await?.json().await?;
+
+        // Find the last_modified for our collection
+        let last_modified = monitor_response["data"]
+            .as_array()
+            .context("No data array in monitor response")?
+            .iter()
+            .find(|record| {
+                record["collection"].as_str() == Some(&collection)
+                    && record["bucket"].as_str() == Some("main")
```

Comment:
You will probably have to take into account that we have other buckets: `blocklists` and `security-state`.

We usually refer to collection as `{bucket_id}/{collection_id}` (eg. `blocklists/addons-bloomfilters`, `main/cfr`, `security-state/onecrl`, ...) and then split it into bucket and collection names

Reply:
Understand now! Thanks! Which collections should we fetch here? For example this one `anti-tracking-url-decoration` exists in the `main` and `main-preview` bucket. 

Can you pass over a list of `collections` we want to have?

Reply:
> Which collections should we fetch here? For example this one anti-tracking-url-decoration exists in the main and main-preview bucket.

We don't download collections from `*-preview` buckets. Only published data ends up in dumps. [More info](https://remote-settings.readthedocs.io/en/latest/tutorial-multi-signoff.html#introduction)

>  How do we know which bucket we want to get the collection from?

You could use the list returned from `/buckets/monitor/collections/changes` to know in which bucket is which collection.
But, on desktop, we scan the dump folder: https://searchfox.org/mozilla-central/source/services/settings/dumps, and get the bucket name from the path: `services/settings/dumps/{bucket}/{collection}.json`